### PR TITLE
kotatogram-desktop: 1.2 -> 1.3.5

### PR DESCRIPTION
**Pull request by thiagokokada**
###### Motivation for this change

Many new features and bugfixes: https://github.com/kotatogram/kotatogram-desktop/releases/tag/k1.3.5

Based on AUR recipe: https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=kotatogram-desktop

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/instant-messengers/telegram/kotatogram-desktop/default.nix [hunk 1]
```diff
@@ -23,16 +23,20 @@ mkDerivation rec {
...
     "--prefix PATH : ${xdg-utils}/bin"
   ];
 
   cmakeFlags = [
+    "-DCMAKE_BUILD_TYPE=Release"
     "-DTDESKTOP_API_TEST=ON"
+    "-DDESKTOP_APP_USE_PACKAGED_GSL=OFF"
     "-DDESKTOP_APP_USE_PACKAGED_RLOTTIE=OFF"
     "-DDESKTOP_APP_USE_PACKAGED_VARIANT=OFF"
+    "-DTDESKTOP_USE_PACKAGED_TGVOIP=OFF"
```
Comment (thiagokokada) [inline pkgs/applications/networking/instant-messengers/telegram/kotatogram-desktop/default.nix:39]: Took it from here: https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=kotatogram-desktop#n60

File: pkgs/applications/networking/instant-messengers/telegram/kotatogram-desktop/default.nix [hunk 2]
```diff
@@ -23,16 +23,20 @@ mkDerivation rec {
...
     zlib minizip openssl hunspell libtgvoip microsoft_gsl tl-expected range-v3
+    libopus alsaLib libpulseaudio
   ];
 
   qtWrapperArgs = [
     "--prefix PATH : ${xdg-utils}/bin"
   ];
 
   cmakeFlags = [
+    "-DCMAKE_BUILD_TYPE=Release"
```
Comment (ilya-fedin) [inline pkgs/applications/networking/instant-messengers/telegram/kotatogram-desktop/default.nix:34]: Isn't it automatically set by Nix?

**Comment by ilya-fedin** [comment]
I planned to update as soon as there will be a stable release based on a newer tdesktop version since build process changed a lot (e.g. there are no PACKAGED options for each library).

There are two problems though:
1. 1.3.5 is based on a tdesktop version with a big fonts bug, so update should be to 1.3.4 or some patching is needed.
2. 1.3.5 is based on a tdesktop version which started to use GTK for mandatory features, so it should be enabled (and patching must be done in order to let if find GTK with dlopen).

File: pkgs/applications/networking/instant-messengers/telegram/kotatogram-desktop/default.nix (hunk 2 above)
Comment (thiagokokada) [reply pkgs/applications/networking/instant-messengers/telegram/kotatogram-desktop/default.nix:34]: Looking at other derivations, doesn't seem so.

File: pkgs/applications/networking/instant-messengers/telegram/kotatogram-desktop/default.nix [hunk 3]
```diff
@@ -1,38 +1,42 @@
 { mkDerivation, lib, fetchFromGitHub, pkg-config, python3, cmake, ninja
 , qtbase, qtimageformats, libdbusmenu, hunspell, xdg-utils, ffmpeg_3, openalSoft
-, lzma, lz4, xxHash, zlib, minizip, openssl, libtgvoip, microsoft_gsl, tl-expected
```
Comment (thiagokokada) [inline pkgs/applications/networking/instant-messengers/telegram/kotatogram-desktop/default.nix:3]: Doesn't seem to be used with `-DDESKTOP_APP_USE_PACKAGED_GSL=OFF` (BTW, it doesn't build without this flag).

**Comment by thiagokokada** [comment]
> I planned to update as soon as there will be a stable release based on a newer tdesktop version since build process changed a lot (e.g. there are no PACKAGED options for each library).
> 
> There are two problems though:
> 
>     1. 1.3.5 is based on a tdesktop version with a big fonts bug, so update should be to 1.3.4 or some patching is needed.
> 
>     2. 1.3.5 is based on a tdesktop version which started to use GTK for mandatory features, so it should be enabled (and patching must be done in order to let it find GTK with dlopen).

Huh... Makes sense. Ok, closing this.

**Pull request closed without merging**